### DISCUSSION TITLE
Report offline updates to hawkbit

### DISCUSSION THREAD

**Discussion by Sourabh1107** [rauc/rauc-hawkbit-updater#120] [Q&A]
Hello,

I wanted to know whether does rauc and rauc-hawkbit-updater support reporting updates to hawkbit which are done via offline using USB stick, etc. Can you please let me know if this is already available and how to use it?

Thanks in advance

**Comment by Bastian-Krause** [comment]
No, offline updates are not supported, see #17.

rauc-hawkbit-updater communicates with hawkBit via its [DDI API](https://www.eclipse.org/hawkbit/apis/ddi_api/). This API does not support offline updates, yet. There was talk about adding offline update functionality [here](https://github.com/eclipse/hawkbit/issues/242#issuecomment-314399692).

The [Management API](https://www.eclipse.org/hawkbit/apis/management_api/) endpoint [/rest/v1/distributionsets/{distributionSetId}/assignedTargets](https://www.eclipse.org/hawkbit/rest-api/distributionsets-api-guide/#_post_rest_v1_distributionsets_distributionsetid_assignedtargets) supports the request query parameter `offline`. If you are bootstrapping devices in a factory, you can use `script/hawkbit_mgmt.py` to talk to the Management API.

**Warning:** The Management API is used to _manage_ hawkBit, e.g. add deployments and assign them to targets. It offers the same functionality as hawkBit's web interface. So you should not allow the target (that should be updated via USB mass storage) to access this API. This is obviously not an API rauc-hawkbit-updater should use.

**Comment by UVV-gh** [comment]


**Reply by Bastian-Krause** [reply]
I think this is the wrong discussion: doesn't this belong in #156?

**Reply by UVV-gh** [reply]
My bad :( I had several tabs open and accidentally commented in the wrong one :)
